Updated docker-on-docker build-notes.

javabrett opened the pull request:
These are now more in-line with wiki instructions. Also removes
broken/deprecated make target `test-unit`.

This change is main `Dockerfile` comments only.  These have fallen out of sync with contribution instructions in the wiki e.g. https://github.com/moby/moby/blob/master/docs/contributing/set-up-dev-env.md and https://github.com/moby/moby/blob/master/docs/contributing/test.md .

I focused on in-container development-workflow.

I left the release-block as-is, as I am unfamiliar with that.

javabrett commented:
Any feedback for this doc-change?

vdemeester commented on Dockerfile:
```diff
@@ -2,14 +2,21 @@
+# root@a8b2885ab900:/go/src/github.com/docker/docker# make install
+#
+# # Run unit tests
+# root@a8b2885ab900:/go/src/github.com/docker/docker# hack/make/test
```
seems to be `hack/test/unit` here 👼

javabrett replied:
Done.

dnephin commented on Dockerfile:
```diff
@@ -2,14 +2,21 @@
-# # Mount your source in an interactive container for quick testing:
-# docker run -v `pwd`:/go/src/github.com/docker/docker --privileged -i -t docker bash
+# # Make a dockerd binary.
+# root@a8b2885ab900:/go/src/github.com/docker/docker# hack/make.sh binary
```
Remove the prompt from this and the examples below

```
hack/make.sh binary
```

mistyhacks replied:
Do you need to show that it is a root prompt, though?

dnephin replied:
It should be inside the container, so it's going to be a root prompt. I think the important detail is to document that this is "inside the shell from the previous step".

javabrett replied:
Done - although I removed the `root@` and retained the `#` ... can remove the `#` too but it looked OK to me.  Added a preceding note about the following commands being run from within the container.

dnephin commented on Dockerfile:
```diff
@@ -2,14 +2,21 @@
+# root@a8b2885ab900:/go/src/github.com/docker/docker# hack/make/test
+#
+# # Run tests e.g. integration, py
+# root@a8b2885ab900:/go/src/github.com/docker/docker# hack/make.sh dynbinary binary cross test-integration test-docker-py
```
```
hack/make.sh binary test-integration test-docker-py
```

javabrett replied:
Done.

javabrett committed:
Updated docker-on-docker build-notes.

These are now more in-line with wiki instructions. Also removes
broken/deprecated make target test-unit.

Signed-off-by: Brett Randall <javabrett@gmail.com>

vdemeester reviewed (approved):
LGTM 🐯
